### PR TITLE
BUG ? in sol grad

### PR DESCRIPTION
**Pull request by agramfort**
@Eric89GXL can you have a look

bug reported by @joewalter

### CODE REVIEW

File: mne/forward/forward.py [hunk 1]
```diff
@@ -642,6 +642,7 @@ def convert_forward_solution(fwd, surf_ori=False, force_fixed=False,
             x = sparse.block_diag([surf_rot] * 3)
             fwd['sol_grad']['data'] = fwd['_orig_sol_grad'] * x  # dot prod
             fwd['sol_grad']['ncol'] = 3 * fwd['nsource']
```
Comment (agramfort) [inline mne/forward/forward.py]: @joewalter says it should be:

```
fwd['sol_grad']['ncol'] = 3 * 3 * fwd['nsource']
```

Comment (joewalter) [inline mne/forward/forward.py]: I can also take care of this fix in #4372

Comment (joewalter) [inline mne/forward/forward.py]: If it is a bug, it needs to be changed in line 656, too.

Comment (joewalter) [inline mne/forward/forward.py]: For testing, read a fwd generated with `--grad` option with `surf_ori=False` and `force_fixed=False`, and read `fwd['sol_grad']['ncol']`.

Then convert it e.g. to `surf_ori=True`, and read `fwd['sol_grad']['ncol']` again.

**Comment by agramfort** [comment]
so you confirm the bug is not present in the C code?

**Comment by joewalter** [comment]
There is a discrepancy before and after manipulating the forward with convert_forward_solution.

**Comment by agramfort** [comment]
can you write the test that fail to we do it the TDD way?

cf the PR with assert I did.

**Comment by joewalter** [comment]
Sorry, this test only works in my convert_fwd branch in #4372, as we always apply convert_forward_solution in read_forward_solution on master.

**Comment by Eric89GXL** [comment]
Also in the test is it easy to create a forward with MNE-C that we can treat as "ground truth" to compare our Python result to?

**Comment by joewalter** [comment]
Can I create a forward with MNE-C and have it in mne-python without using our read function for io?

**Comment by Eric89GXL** [comment]
> Can I create a forward with MNE-C and have it in mne-python without using our read function for io?

No. I assumed the bug was in the conversion routines, but if they are in the reading routines then it's not worth it.

**Comment by joewalter** [comment]
> No. I assumed the bug was in the conversion routines, but if they are in the reading routines then it's not worth it.

We can add the test in #4372, where we separate reading and converting forward solutions. Shall we try to merge #4372 before this one (or handle the bug there, which I think I already did)?

**Comment by Eric89GXL** [comment]
Let's get #4372 in and make additional fixes later as necessary. I think that one should be good to go anyway

**Comment by larsoner** [comment]
I think this was fixed in #4372 / #4620, feel free to reopen if not

**Pull request closed without merging**